### DISCUSSION TITLE
How to replace some values inside a specific tag in a index.html file using a cypress automation test

### DISCUSSION THREAD

**Discussion by ShanikaWickramasinghe** [cypress-io/cypress#15561] [Questions and Help] [answered]
Currently, I am working with cypress test automation. I have an index.html file inside cypress/downloads/sample/index.html. The index.html file has the below structure.

```
<html>
<head>
  <style type="text/css">
  </style>
</head>
<body>
<script src="https://test"></script>
<script src="https://test2"></script>

<!-- sample Config -->
  <script>
        const authConfig = {
            clientID: "sample value",
            signInRedirectURL: "sample value",
            serverOrigin: "sample value"
        };
    </script>
</body>
</html>
```
I will get copy the content inside the script tag which will be displayed in an application UI using a button click for the copy operation so the copied content will be as below 

```
<script>
        const authConfig = {
            clientID: "abcd1234",
            signInRedirectURL: "https://localhost:8000",
            serverOrigin: "https://localhost:8000"
        };
    </script>
```
```
static CopyConfigs() {

        cy.get(COPY_ICON).click();
        cy.wait(7000);
    }

static pasteoperation(Location){
            cy.task("getClipboard").then((value) => {
                cy.writeFile(Location + "/sample/index.html", value);
            });
    }
```
writeFile will replace the whole index.html file content

**Need help on**  
I want to replace(paste) the content copied to the index.html file by replacing only the script tag(related to authConfig) content inside the index.html file without replacing the whole index.html file content. So that the values for clientID, signInRedirectURL, and serverOrigin inside the script tag of index.html file will have the new values. Is it possible to read the index.html file and capture the script tag and replace the <script> tag content using cypress automation? Appreciate your support on how to achieve this

**Comment by bahmutov** [comment, marked as answer]
- you can use `cy.intercept` to rewrite / replace content of the HTML file, see the HTML example spec in https://github.com/cypress-io/cypress-example-recipes/tree/master/examples/stubbing-spying__intercept

**Reply by ShanikaWickramasinghe** [reply]
@bahmutov Thanks for looking into this. BTW can we use cy.intercept method to paste a copied content by reading a file inside my cypress downloads folder under a specific tag of the file reside inside the cypress project. (cypress/downloads/sample/index.html). I don't want to replace the whole file content. The config that I copy from the UI, I want to paste it under <script> const authConfig  </script> tag of my index.html file which resides inside downloads.

**Comment by bahmutov** [comment]
Yes you can. Just load the JS file as text beforehand and insert. See our html modification example in the cy.intercept recipe

**Reply by ShanikaWickramasinghe** [reply]
@bahmutov Thank You for your time. Can You please provide me the exact link for the specific spec file. It seems there are several spec files in the link provided in the above reply.

**Comment by bahmutov** [comment]
Look at this one

html-css-spec.js modifies the page itself and changes the CSS requested by the page
